File: legacy/graphql-java/src/main/java/com/example/demo/gql/scalars/Scalars.java
package com.example.demo.gql.scalars;

import graphql.schema.GraphQLScalarType;

public class Scalars {

    public static GraphQLScalarType uuidType() {
        return GraphQLScalarType.newScalar()
                .name("UUID")
                .description("UUID type")
                .coercing(new UUIDScalar())
                .build();
    }

    public static GraphQLScalarType localDateTimeType() {
        return GraphQLScalarType.newScalar()
                .name("LocalDateTime")
                .description("LocalDateTime type")
                .coercing(new LocalDateTimeScalar())
                .build();
    }
}

File: graphql-kotlin/src/main/kotlin/com/example/demo/gql/scalars/Scalars.kt
package com.example.demo.gql.scalars

import graphql.schema.GraphQLScalarType


object Scalars {
    val uuidType: GraphQLScalarType = GraphQLScalarType.newScalar()
        .name("UUID")
        .description("UUID type")
        .coercing(UUIDScalar())
        .build()
    val localDateTimeType: GraphQLScalarType = GraphQLScalarType.newScalar()
        .name("LocalDateTime")
        .description("LocalDateTime type")
        .coercing(LocalDateTimeScalar())
        .build()

}

